pathway-analysis/ssgsea_example.Rmd

```diff
@@ -0,0 +1,392 @@
+  dir.create("plots")
+}
+
+if (!dir.exists("results")) {
+  dir.create("results")
+}
+```
+
+## Read in refine.bio data
+
+The gene expression matrix of the dataset we'll be working with is too large 
+to be tracked with git without compression, so we need to unzip it if we have
+not already.
+This compressed folder contains both the expression matrix and the metadata.
+
+```{r}
+# if the unzipped folder does not exist
+if (!dir.exists(file.path("data", "GSE75574"))) {
+  compressed_file <- file.path("data", "GSE75574.zip")
+  unzip(compressed_file, exdir = "data")
+}
+```
+
+### Expression data
+
+Because we used this in another notebook, it's possible that you already have
+a local copy of the prepared expression matrix.
+If a local copy exists, we don't want to go through the steps to reprocess it,
+so we can check for this first.
+
+If not, we'll process the data in the following ways:
+
+* Convert the Ensembl gene IDs that are used in refine.bio data to gene symbols.
+* If duplicate gene symbols exist, we'll collapse to the mean value for that 
+gene symbol.
+
+```{r}
+# check for the existence of the file that would have been generated in the
+# other notebook
+symbol_file <- file.path("data", "GSE75574", "GSE75574_symbols.tsv")
+if (file.exists(symbol_file)) {
+  symbol_exprs_df <- readr::read_tsv(symbol_file, progress = FALSE)
+} else {
+  
+  # read in the refine.bio expression data
+  expression_file <- file.path("data", "GSE75574", "GSE75574.tsv")
+  exprs_df <- readr::read_tsv(expression_file, progress = FALSE)
+  # readr will fill in the first column, which contains the Ensembl gene IDs, as
+  # 'X1'
+  colnames(exprs_df)[1] <- "ENSEMBL"
+  
+  # using the default behavior for 1:many mappings here -- select the first one
+  symbol_mappings <- mapIds(org.Mm.eg.db, keys = exprs_df$ENSEMBL, 
+                          column = "SYMBOL", keytype = "ENSEMBL")
+  
+  # mapIds should give us back IDs in the same order as the keys, but we can
+  # check to be sure
+  if (!(all.equal(names(symbol_mappings), exprs_df$ENSEMBL))) {
+    stop("Gene order is not as expected!")
+  }
+  
+  symbol_exprs_df <- exprs_df %>%
+    # add a new column that contains the gene symbols 
+    # this gets added as the last column
+    dplyr::mutate(SYMBOL = symbol_mappings) %>%
+    # drop the Ensembl gene IDs
+    dplyr::select(-ENSEMBL) %>%
+    # reorder such that the gene symbols are in the first column
+    dplyr::select(SYMBOL, dplyr::everything()) %>%
+    # drop any genes without a gene symbol
+    dplyr::filter(!is.na(SYMBOL))
+  
+  # if there are any duplicate gene symbols, collapse to the mean value
+  if (any(duplicated(symbol_exprs_df$SYMBOL))) {
+    symbol_exprs_df <- symbol_exprs_df %>%
+      dplyr::group_by(SYMBOL) %>%
+      dplyr::summarise_all(mean)
+  }
+  
+  # write to file
+  readr::write_tsv(symbol_exprs_df, symbol_file)
+}
+```
+
+### Metadata 
+
+```{r}
+metadata_file <- file.path("data", "GSE75574", "metadata_GSE75574.tsv")
```

Comment:
I don't remember, is the metadata not saved so we can't try to load it like we just did with the edited expression data?

Reply:
The metadata does not get saved in the single dataset example.